slave/slave.go

```diff
@@ -37,13 +37,14 @@ func (s *Slave) Run() {
 	for {
 		select {
 		case task := <-s.addTaskChan:
-			addTask(task, s)
+			s.addTask(task)
+			go s.Execute(task)
```

Comment:
just one thing - although this won't cause a race condition because you're only using the channel in the go routine, but if we change the `Execute` method in the future and access any of the fields then we'll have problems. so to be extra safe, it's best to write `execute(task, slave.successChan, slave.failChan)` so that we're certain it's not modifying `s`.